haystack/components/converters/csv.py

```diff
@@ -102,3 +105,231 @@ def run(
+
+    def _clean_blocks(self, blocks: List[List[List[str]]]) -> List[List[List[str]]]:
+        """
+        Existing row-based cleaning (trim leading/trailing empty rows).
+        """
+        cleaned = []
+        for block in blocks:
+            if not self.trim_empty_rows:
+                cleaned_block = block.copy()
+            else:
+                start, end = 0, len(block)
+                while start < end and all(cell.strip() == "" for cell in block[start]):
+                    start += 1
+                while end > start and all(cell.strip() == "" for cell in block[end - 1]):
+                    end -= 1
+                cleaned_block = block[start:end]
+
+            if cleaned_block:
+                cleaned.append(cleaned_block)
+        return cleaned
+
+    def _clean_single_block_2d(self, block_2d: List[List[str]]) -> List[List[str]]:
+        """
+        For a 2D block remove empty top/bottom rows, and empty left/right columns.
+        """
+        if not block_2d:
+            return []
+
+        if self.trim_empty_rows:
+            start_r, end_r = 0, len(block_2d)
+            while start_r < end_r and all(cell.strip() == "" for cell in block_2d[start_r]):
+                start_r += 1
+            while end_r > start_r and all(cell.strip() == "" for cell in block_2d[end_r - 1]):
+                end_r -= 1
+            block_2d = block_2d[start_r:end_r]
+
+        if not block_2d:
+            return []
+
+        min_col, max_col = 0, len(block_2d[0])
+        while min_col < max_col:
+            if all(row[min_col].strip() == "" for row in block_2d):
+                min_col += 1
+            else:
+                break
+        while max_col > min_col:
+            if all(row[max_col - 1].strip() == "" for row in block_2d):
+                max_col -= 1
+            else:
+                break
+
+        if min_col >= max_col:
+            return []
+
+        cleaned = [row[min_col:max_col] for row in block_2d]
+        return cleaned
+
+    def _block_to_csv(self, block: List[List[str]]) -> str:
+        """Converts a 2D block of rows into CSV text."""
+        output = io.StringIO()
+        writer = csv.writer(output, delimiter=self.delimiter, quotechar=self.quotechar, quoting=csv.QUOTE_MINIMAL)
+        writer.writerows(block)
+        return output.getvalue().strip()
+
+    @component.output_types(documents=List[Document])
+    def run(self, documents: List[Document]) -> Dict[str, List[Document]]:
+        """Processes documents to split CSV content."""
+        split_docs: List[Document] = []
+        for doc in documents:
+            if not doc.content:
+                continue
+            try:
+                blocks = self._split_csv_content(doc.content)
+            except Exception as e:
+                if not self.skip_errors:
+                    raise
+                warnings.warn(f"Skipping document {doc.id}: {str(e)}", UserWarning)
+                continue
```

Comment:
Also I'd rather not fully skip the document, but instead pass it along and warn the user that it was unable to be split.